src/js/common/schemaform/actions.js

```diff
@@ -0,0 +1,95 @@
+import { flatternFormData } from './helpers';
+import environment from '../helpers/environment.js';
+
+export const SET_EDIT_MODE = 'SET_EDIT_MODE';
+export const SET_DATA = 'SET_DATA';
+export const SET_VALID = 'SET_VALID';
+export const SET_PRIVACY_AGREEMENT = 'SET_PRIVACY_AGREEMENT';
+export const SET_SUBMISSION = 'SET_SUBMISSION';
+export const SET_SUBMITTED = 'SET_SUBMITTED';
+
+export function setData(page, data) {
+  return {
+    type: SET_DATA,
+    data,
+    page
+  };
+}
+
+export function setValid(page, valid) {
+  return {
+    type: SET_VALID,
+    page,
+    valid
+  };
+}
+
+export function setEditMode(page, edit) {
+  return {
+    type: SET_EDIT_MODE,
+    edit,
+    page
+  };
+}
+
+export function setSubmission(field, value) {
+  return {
+    type: SET_SUBMISSION,
+    field,
+    value
+  };
+}
+
+export function setPrivacyAgreement(privacyAgreementAccepted) {
+  return {
+    type: SET_PRIVACY_AGREEMENT,
+    privacyAgreementAccepted
+  };
+}
+
+export function setSubmitted(response) {
+  return {
+    type: SET_SUBMITTED,
+    response
+  };
+}
+
+export function submitForm(formConfig, form) {
```

Comment:
This whole function should be generic, but there's some edu specific stuff in here for now.